src/lib/Scenes/Artwork/Components/RequestConditionReport.tsx

```diff
@@ -0,0 +1,186 @@
+  showErrorModal: boolean
+  errorModalText: string
+}
+
+@track({
+  action_name: Schema.ActionNames.RequestConditionReport,
+  context_module: Schema.ContextModules.ArtworkDetails,
+})
+export class RequestConditionReport extends React.Component<RequestConditionReportProps, State> {
+  state = {
+    requestingConditionReport: false,
+    showConditionReportRequestedModal: false,
+    showErrorModal: false,
+    errorModalText: "",
+  }
+
+  requestConditionReport = () => {
+    const { artwork } = this.props
+    return new Promise(async (resolve, reject) => {
+      commitMutation<RequestConditionReportMutation>(defaultEnvironment, {
+        onCompleted: data => {
+          resolve(data)
+        },
+        onError: error => {
+          reject(error)
+        },
+        mutation: graphql`
+          mutation RequestConditionReportMutation($input: RequestConditionReportInput!) {
+            requestConditionReport(input: $input) {
+              conditionReportRequest {
+                internalID
+              }
+            }
+          }
+        `,
+        variables: {
+          input: { saleArtworkID: artwork.saleArtwork.internalID },
+        },
+      })
+    })
+  }
+
+  @track({
+    action_type: Schema.ActionTypes.Fail,
+  })
+  presentErrorModal(errors: Error | ReadonlyArray<PayloadError>, mutationMessage: string) {
+    console.error("RequestConditionReport.tsx", errors)
+    const errorMessage = mutationMessage || "There was a problem processing your request. Please try again."
+    this.setState({ showErrorModal: true, errorModalText: errorMessage, requestingConditionReport: false })
+  }
+
+  @track({
+    action_type: Schema.ActionTypes.Success,
+  })
+  presentSuccessModal() {
+    this.setState({ showConditionReportRequestedModal: true, requestingConditionReport: false })
+  }
+
+  @track({
+    action_type: Schema.ActionTypes.Tap,
+  })
+  handleRequestConditionReportTap() {
+    this.setState({ requestingConditionReport: true })
+
+    this.requestConditionReport()
+      .then(data => {
+        const theData = data as any
+        if (theData.requestConditionReport) {
+          this.presentSuccessModal()
+        } else {
+          this.presentErrorModal(null, null)
+        }
+      })
+      .catch(error => {
+        this.presentErrorModal(error, null)
+      })
+  }
+
+  closeModals() {
+    this.setState({ showConditionReportRequestedModal: false, showErrorModal: false })
+  }
+
+  render() {
+    const { me } = this.props
+    const { requestingConditionReport, showErrorModal, errorModalText, showConditionReportRequestedModal } = this.state
+
+    return (
+      <View>
+        <Button
+          mt={1}
+          size="small"
+          variant="secondaryGray"
+          loading={requestingConditionReport}
+          onPress={this.handleRequestConditionReportTap.bind(this)}
+        >
+          Request condition report
+        </Button>
+        <Flex height={0}>
```

Comment:
is this Flex wrapper required? my understanding is that Modals do not affect layout

Reply:
The margin on containing view in our implementation of the modal is messing with layout: https://github.com/artsy/emission/blob/master/src/lib/Components/Modal.tsx#L54 This seems like a bug so probably want to remove but doing so broke a couple layout tests for other components and seemed outside scope of this PR. So I went with this workaround instead. 

![requestConditionReportGap](https://user-images.githubusercontent.com/49686530/73963341-1629d000-48de-11ea-80ca-e98226054e5f.png)

Reply:
ah ok, makes sense then